packages/flutter/lib/src/material/pickers/date_range_picker_dialog.dart

```diff
@@ -0,0 +1,678 @@
+        );
+        size = orientation == Orientation.portrait ? _inputPortraitDialogSize : _inputLandscapeDialogSize;
+        insetPadding = const EdgeInsets.symmetric(horizontal: 16.0, vertical: 24.0);
+        shape = const RoundedRectangleBorder(
+          borderRadius: BorderRadius.all(Radius.circular(4.0))
+        );
+        break;
+    }
+
+    return Dialog(
+      child: AnimatedContainer(
+        width: size.width,
+        height: size.height,
+        duration: _dialogSizeAnimationDuration,
+        curve: Curves.easeIn,
+        child: MediaQuery(
+          data: MediaQuery.of(context).copyWith(
+            textScaleFactor: textScaleFactor,
+          ),
+          child: Builder(builder: (BuildContext context) {
+            return contents;
+          }),
+        ),
+      ),
+      insetPadding: insetPadding,
+      shape: shape,
+      clipBehavior: Clip.antiAlias,
+      elevation: 24.0,
+    );
+  }
+}
+
+class _CalendarRangePickerDialog extends StatelessWidget {
+  const _CalendarRangePickerDialog({
+    Key key,
+    @required this.selectedStartDate,
+    @required this.selectedEndDate,
+    @required this.firstDate,
+    @required this.lastDate,
+    @required this.currentDate,
+    @required this.onStartDateChanged,
+    @required this.onEndDateChanged,
+    @required this.onConfirm,
+    @required this.onCancel,
+    @required this.onToggleEntryMode,
+    @required this.confirmText,
+    @required this.helpText,
+  }) : super(key: key);
+
+  final DateTime selectedStartDate;
+  final DateTime selectedEndDate;
+  final DateTime firstDate;
+  final DateTime lastDate;
+  final DateTime currentDate;
+  final ValueChanged<DateTime> onStartDateChanged;
+  final ValueChanged<DateTime> onEndDateChanged;
+  final VoidCallback onConfirm;
+  final VoidCallback onCancel;
+  final VoidCallback onToggleEntryMode;
+  final String confirmText;
+  final String helpText;
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final ColorScheme colorScheme = theme.colorScheme;
+    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
+    final Orientation orientation = MediaQuery.of(context).orientation;
+    final TextTheme textTheme = theme.textTheme;
+    final Color headerForeground = colorScheme.brightness == Brightness.light
+      ? colorScheme.onPrimary
+      : colorScheme.onSurface;
+    final Color headerDisabledForeground = headerForeground.withOpacity(0.38);
+    final String startDateText = utils.formatRangeStartDate(localizations, selectedStartDate, selectedEndDate);
+    final String endDateText = utils.formatRangeEndDate(localizations, selectedStartDate, selectedEndDate, DateTime.now());
+    final TextStyle headlineStyle = textTheme.headline5;
+    final TextStyle startDateStyle = headlineStyle?.apply(color: selectedStartDate != null ? headerForeground : headerDisabledForeground);
```

Comment:
You could probably format lines like this like so to be closer to 80 chars:
```suggestion
    final TextStyle startDateStyle = headlineStyle?.apply(
      color: selectedStartDate != null ? headerForeground : headerDisabledForeground,
    );
```